frontend/its-our-call/src/App.tsx

```diff
@@ -1,25 +1,59 @@
 import * as React from 'react';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
 import * as pages from './pages';
+import * as urls from './pages/urls';
 import './App.css';
 
 import Header from './components/Header';
+import { SearchOverlay, MainMenu } from './components';
 
-import * as urls from './pages/urls';
+interface AppProps { }
+
+interface AppState {
+  searchHidden: boolean;
+  menuHidden: boolean;
+}
+
+class App extends React.Component<AppProps, AppState> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {
+      searchHidden: true,
+      menuHidden: true,
+    };
+  }
+
+  handleSearchClick = () => {
+    this.setState({
+      searchHidden: !this.state.searchHidden,
+      menuHidden: true,
+    });
+  }
+
+  handleMenuClick = () => {
+    this.setState({
+      searchHidden: true,
+      menuHidden: !this.state.menuHidden,
```

Comment:
searchHidden and menuHidden might be better as one enum (since search and menu will probably never be shown together) but they can stay separate for now.

Personally I use `hidden` flags as optional booleans so you don't have to explicitly set them but the code works as it is.

Reply:
Makes sense. Using a single tri-state variable would prevent any bugs with `searchHidden` and `menuHidden` both being true (assuming that's not desired).